test/pylib/scylla_server.py

```diff
@@ -0,0 +1,307 @@
+
+#
+# Put all Scylla options in a template file. Sic: if you make a typo in the
+# configuration file, Scylla will boot fine and ignore the setting.
+# Always check the error log after modifying the template.
+#
+SCYLLA_CONF_TEMPLATE = """cluster_name: {cluster_name}
+developer_mode: true
+
+# Allow testing experimental features. Following issue #9467, we need
+# to add here specific experimental features as they are introduced.
+
+enable_user_defined_functions: true
+experimental: true
+experimental_features:
+    - raft
+    - udf
+
+data_file_directories:
+    - {workdir}/data
+commitlog_directory: {workdir}/commitlog
+hints_directory: {workdir}/hints
+view_hints_directory: {workdir}/view_hints
+
+listen_address: {host}
+rpc_address: {host}
+api_address: {host}
+prometheus_address: {host}
+alternator_address: {host}
+
+seed_provider:
+    - class_name: org.apache.cassandra.locator.simple_seed_provider
+      parameters:
+          - seeds: {seeds}
+
+skip_wait_for_gossip_to_settle: 0
+ring_delay_ms: 0
+num_tokens: 16
+flush_schema_tables_after_modification: false
+auto_snapshot: false
+
+# Significantly increase default timeouts to allow running tests
+# on a very slow setup (but without network losses). Note that these
+# are server-side timeouts: The client should also avoid timing out
+# its own requests - for this reason we increase the CQL driver's
+# client-side timeout in conftest.py.
+
+range_request_timeout_in_ms: 300000
+read_request_timeout_in_ms: 300000
+counter_write_request_timeout_in_ms: 300000
+cas_contention_timeout_in_ms: 300000
+truncate_request_timeout_in_ms: 300000
+write_request_timeout_in_ms: 300000
+request_timeout_in_ms: 300000
+
+# Set up authentication in order to allow testing this module
+# and other modules dependent on it: e.g. service levels
+
+authenticator: PasswordAuthenticator
+strict_allow_filtering: true
+"""
+
+# Seastar options can not be passed through scylla.yaml, use command line
+# for them. Keep everything else in the configuration file to make
+# it easier to restart. Sic: if you make a typo on the command line,
+# Scylla refuses to boot.
+SCYLLA_CMDLINE_OPTIONS = [
+    '--smp', '2',
+    '-m', '1G',
+    '--collectd', '0',
+    '--overprovisioned',
+    '--max-networking-io-control-blocks', '100',
+    '--unsafe-bypass-fsync', '1',
+    '--kernel-page-cache', '1',
+]
+
+
+class ScyllaServer:
+    # Regular expression to scan the log
+    STARTUP_MSG_RE = re.compile(".*Scylla.*initialization completed")
+
+    def __init__(self, exe=None, vardir=None, host_registry=None,
+                 cluster_name=None, seed=None, cmdline_options=None) -> None:
+        self.exe = pathlib.Path(exe).resolve()
+        self.vardir = pathlib.Path(vardir)
+        self.host_registry = host_registry
+        self.cmdline_options = cmdline_options
+        self.cfg = {
+            "cluster_name": cluster_name,
+            "host": None,
+            "seeds": seed,
+            "workdir": None,
```

Comment:
It seems like you often use these keys inside the class below, so they should be member variables (to preserve the types for instance), and you could construct the dictionary when you actually need it from these member variables.